compiler/semexprs.nim

```diff
@@ -1647,6 +1647,9 @@ proc semAsgn(c: PContext, n: PNode; mode=asgnNormal): PNode =
       # unfortunately we need to rewrite ``(x, y) = foo()`` already here so
       # that overloading of the assignment operator still works. Usually we
       # prefer to do these rewritings in transf.nim:
+      a = semExprWithType(c, a)
+      let rhs = semExprWithType(c, n.sons[1])
+      n.sons[1] = fitNode(c, a.typ, rhs, rhs.info)
```

Comment:
Sorry, but I don't like this solution. Calling ``semExpr`` twice ... there has to be a better way.

Reply:
I think its possible to remove the extra semexpr on either the lhs or rhs, but I can't come up with a way to remove both without replicating assignment logic with resugaring for error messages.